forwards_paper.tex

```diff
@@ -260,79 +260,93 @@ \subsection*{Tree sequences}
+describing the recorded edges.
+}
+
+\algstep{W1.}{Initialisation.}{
+ For $0 \leq j < N$, set $P_j \leftarrow j$ and $\tau_j \leftarrow T$.
+Set $E\leftarrow \emptyset$ and $n \leftarrow N$.
+}
+
+\algstep{W2.}{Generation loop.}{ If $t = 0$ terminate.
+Set $t \leftarrow t - 1$ and $j \leftarrow 0$.
+}
+
+\algstep{W3.}{Individual loop.}{ If $j = N$ set $P \leftarrow P'$ and go to \algref{W2}.
+}
+
+\algstep{W4.}{Mortality.}{ If $\randomuniform([0, 1)) > \delta$, set $P'_j \leftarrow P_j$
+    and go to \algref{W7}.
+}
+
+\algstep{W5.}{New node.}{Set $P'_j \leftarrow n$ and $\tau_n \leftarrow t$.
+}
+
+\algstep{W6.}{Choose parents.}{Set $a \leftarrow \randomuniform(\{0, \dots, N - 1\})$,
+    $b \leftarrow \randomuniform(\{0, \dots, N - 1\})$ and $x \leftarrow \randomuniform([0, L))$.
+}
+
+\algstep{W7.}{Record edges}{Set $E \leftarrow E \cup \left\{ (0, x, P_a, n), (x, L, P_b, n) \right\}$
+    and $n \leftarrow n + 1$.
+}
+
+\algstep{W8.}{Loop tail}{Set $j \leftarrow j + 1$ and go to \algref{W4}.
+}
+
+\jk{I haven't checked this carefully. If we used 1-based indexes it might make things a bit
+simpler}.
+\end{taocpalg}
+
+Algorithm~\algref{W} is a very simple implementation of a Wright--Fisher model
+intended to illustrate how we can record the nodes and edges of a tree sequence
+forwards in time. We begin in~\algref{W1} by allocating our initial population $P$
+and creating $N$ nodes with birth time $T$ generations ago, recorded in the
+vector $\tau$. The set $E$ is used to store the edges that we output during the
+simulation, and $n$ is the number of nodes created so far (and so, the ID of
+the next node we create). Steps~\algref{W2} and~\algref{W3} simply loop over
+the generation clock $t$ and individual index $j$. In~\algref{W4} we check if
+an individual $P_j$ has died in this generation. If it has, we replace it in
+steps \algref{W5}--\algref{W7}; if not, we proceed immediately to the next
+individual. When an individual in the population dies, we first allocate
+a new node with ID $n$ in~\algref{W5} and record its birth time. Then,
+in step~\algref{W6} we choose two indexes $a$ and $b$ uniformly (giving us
+parents $P_a$ and $P_b$) and choose a breakpoint $x$. We record the effects of this
+event by storing two new edges: one recording that the parent of node $n$
+from $0$ to $x$ is $P_a$, and another recording that the parent of $n$
+from $x$ to $L$ is $P_b$. We then complete the replacement event by incrementing
+$n$, ready to represent the next new node.
 
 
+\begin{enumerate}
+\item Explain how to add mutational information. We can also do this afterwards on the
+output trees if they are nuetral.
+\item Explain that we can input these trees into an mpsrime tree sequence and they will
+work fine. Perhaps a diagram to show the kind of trees we get? (Multiroots, lots of unary nodes.
+Can use the msprime SVG API to draw these?) We could refer ahead to figure 2
```

Comment:
We could just refer ahead.  But it might be *rad* to have a side-by-side of a pair of pre- and post-simplify trees.